packages/@lwc/template-compiler/src/codegen/index.ts

```diff
@@ -79,14 +79,13 @@ import { format as formatModule } from './formatters/module';
 function transform(codeGen: CodeGen): t.Expression {
     const instrumentation = codeGen.state.config.instrumentation;
     function transformElement(element: BaseElement, slotParentName?: string): t.Expression {
-        const databag = elementDataBag(element, slotParentName);
-        let res: t.Expression;
-
         if (codeGen.staticNodes.has(element) && isElement(element)) {
             // do not process children of static nodes.
             return codeGen.genStaticElement(element as StaticElement, slotParentName);
         }
```

Comment:
It doesn't look like we use the `databag` in `genStaticElement` and we create the estree nodes for static elements separately from anything else in this routine.

This was causing the keys to appear disjoint at times in the rendered template (see fixtures for details).

I moved it to after the check to prevent unneeded parsing.